File: frontend/src/passwordPolicy.test.ts
import { isStrongPassword, passwordPolicyHelper } from './passwordPolicy.js'

function assertEqual(actual: boolean, expected: boolean, label: string) {
  if (actual !== expected) {
    throw new Error(`${label}: got ${actual}, want ${expected}`)
  }
}

assertEqual(isStrongPassword('ChangeMe123!'), true, 'strong password')
assertEqual(isStrongPassword('short1A'), false, 'short password')
assertEqual(isStrongPassword('lowercase123'), false, 'missing uppercase')
assertEqual(isStrongPassword('UPPERCASE123'), false, 'missing lowercase')
assertEqual(isStrongPassword('NoDigitsHere'), false, 'missing digit')

if (!passwordPolicyHelper.includes('10')) {
  throw new Error('password policy helper should mention length')
}

File: backend/internal/auth/password_policy_test.go
package auth

import "testing"

func TestValidatePasswordPolicyAcceptsStrongPassword(t *testing.T) {
	if err := ValidatePasswordPolicy("ChangeMe123!"); err != nil {
		t.Fatalf("ValidatePasswordPolicy returned error for strong password: %v", err)
	}
}

func TestValidatePasswordPolicyRejectsWeakPasswords(t *testing.T) {
	tests := []string{
		"short1A",
		"lowercase123",
		"UPPERCASE123",
		"NoDigitsHere",
		"          ",
	}

	for _, password := range tests {
		if err := ValidatePasswordPolicy(password); err == nil {
			t.Fatalf("ValidatePasswordPolicy(%q) returned nil, want weak password error", password)
		}
	}
}
